primeqa/services/grpc_server/retriever_service.py

```diff
@@ -50,18 +51,28 @@ def GetRetrievers(
         Returns:
             GetRetrieversResponse: List of available retrievers
         """
-        return GetRetrieversResponse(
-            retrievers=[
-                RetrieverComponent(
-                    retriever_id=retriever_id,
-                    parameters=generate_parameters(
-                        retriever, skip=["index_root", "index_name"]
-                    ),
-                )
-                for retriever_id, retriever in RETRIEVERS_REGISTRY.items()
-            ]
-        )
+        retrievers = [
+            RetrieverComponent(
+                retriever_id=retriever_id,
+                parameters=generate_parameters(
+                    retriever, skip=["index_root", "index_name"]
+                ),
+            )
+            for retriever_id, retriever in RETRIEVERS_REGISTRY.items()
+        ]
 
+        # Custom behavior: Add checkpoints as options in checkpoint parameter:
+        try:
+            colbert_component = next(component for component in retrievers if component.retriever_id == 'ColBERTRetriever')
```

Comment:
@hrojek If I am not mistaken in future release @franzmpub might expose retriever based on DPR. That one will also have checkpoint parameter. So maybe  Let's try to move this to a common function in retriever class or ColBERTRetriever class that gets called from to populate options for checkpoint.